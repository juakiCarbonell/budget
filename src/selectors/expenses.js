import moment from "moment";

// Get visible exoenses
const getVisibleExpenses = (expenses, {text, sortBy, startDate, endDate}) => {
  return expenses.filter((expense) => {
    const createdAtMoment = moment(expense.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true ;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true ;
    const textMatch= expense.description.toLowerCase().includes(text.toLowerCase());
    // If any of these variables are false, then the element is filtered from the returned array:
    // All three of those need to be true in order for the expense to not be filtered out.
    return startDateMatch && endDateMatch && textMatch;
  }).sort((a ,b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else if (sortBy === 'amount') {
      return a.amount < b.amount ? 1 : -1;
    }
  });
};

export default getVisibleExpenses;