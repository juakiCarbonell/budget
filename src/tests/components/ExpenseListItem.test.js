import ExpensesListItem from '../../components/ExpenseListItem';
import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem render correctly', () => {
  const wrapper = shallow(<ExpensesListItem {...expenses[1]}/>);
  expect(wrapper).toMatchSnapshot();
});