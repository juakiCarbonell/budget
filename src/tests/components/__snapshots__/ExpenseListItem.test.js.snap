// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem render correctly 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    109600
     - 
    -345600000
  </p>
</div>
`;
